src/solvers/dg_simplices/dg.jl

```diff
@@ -0,0 +1,301 @@
+# !!! warning "Experimental features"
+
+# out <- A*x
+mul_by!(A) = let A = A 
+  @inline (out, x)->matmul!(out, A, x) 
+end
+
+# specialize for SBP operators since `matmul!` doesn't work for `UniformScaling` types.
+mul_by!(A::UniformScaling) = let A = A
+  @inline (out, x)->mul!(out, A, x)
+end
+  
+# # Todo: simplices. Use `matmul!` for the following 2 functions until 5-arg `matmul!` once 
+# the hanging bug is fixed (see https://github.com/JuliaLinearAlgebra/Octavian.jl/issues/103). 
+
+# out <- out + A * x 
+mul_by_accum!(A) = let A = A 
+  @inline (out, x)->mul!(out, A, x, one(eltype(out)), one(eltype(out))) 
+end
+
+#  out <- out + α * A * x 
+mul_by_accum!(A, α) = let A = A 
+  @inline (out, x)->mul!(out, A, x, α, one(eltype(out))) 
+end
+
+const DGWeakForm{Dims, ElemType} = DG{<:RefElemData{Dims, ElemType}, Mortar, 
+                    <:SurfaceIntegralWeakForm,
+                    <:VolumeIntegralWeakForm} where {Mortar}
```

Comment:
Or `VertexMappedDGWeakForm` if you want to distinguish it from strong form variants.

Reply:
> I think the name DGWeakForm is not optimal here, since it's not really an alias for general weak form DG solvers, but very specific in its use of the basis type.

I was thinking this is very much an alias for general DG weak formulations - it supports most common reference elements and general quadrature rules, as well as SBP-DG discretizations. However, since it's not exported, renaming makes sense to avoid possible namespace collisions.

How about `MultiDG`? "Multidimensional" is used in the SBP literature uses this to refer to operators on simplices which are directly constructed given a nodal set (in contrast to DGSEM, where 2D and 3D operators are constructed from 1D operators). It can also reference the fact that this supports multiple types of elements and discretizations.

Reply:
Renaming `DG{<:RefElemData,...}` to `MultiDG{...}`.

Reply:
Thanks, this sounds much better IMHO!